packages/vite/src/node/plugins/html.ts

```diff
@@ -58,11 +58,7 @@ export function htmlInlineScriptProxyPlugin(config: ResolvedConfig): Plugin {
         const file = cleanUrl(id)
         const url = file.replace(normalizePath(config.root), '')
         const result = htmlProxyMap.get(config)!.get(url)![index]
-        if (result) {
-          return result
-        } else {
-          throw new Error(`No matching HTML proxy module found from ${id}`)
-        }
+        return result
```

Comment:
I now see that your initial implementation was right. Sorry for making you do the change here, we should keep the error.
```suggestion
        if (typeof result === 'string') {
          return result
        } else {
          throw new Error(`No matching HTML proxy module found from ${id}`)
        }
```